File: java/MatricuReizinasana.java
package java;
public class MatricuReizinasana {
    public static int[][] matrixMultiply(int[][] a, int[][] b) {
        int rows = a.length;
        int cols = b[0].length;
        int[][] result = new int[rows][cols];

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                for (int k = 0; k < a[0].length; k++) {
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        return result;
    }

    public static void main(String[] args) {
        int size = 100;
        int[][] a = new int[size][size];
        int[][] b = new int[size][size];
        for (int i = 0; i < size; i++) {
            for (int j = 0; j < size; j++) {
                a[i][j] = 1;
                b[i][j] = 2;
            }
        }
        @SuppressWarnings("unused")
        int[][] result = matrixMultiply(a, b);
        System.out.println("Matrix multiplication completed");
    }
}

File: rust/matricu_reizinasana.rs
fn matrix_multiply(a: &Vec<Vec<i32>>, b: &Vec<Vec<i32>>) -> Vec<Vec<i32>> {
    let rows = a.len();
    let cols = b[0].len();
    let mut result = vec![vec![0; cols]; rows];

    for i in 0..rows {
        for j in 0..cols {
            for k in 0..a[0].len() {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    result
}

fn main() {
    let size = 100;
    let a = vec![vec![1; size]; size];
    let b = vec![vec![2; size]; size];
    let _ = matrix_multiply(&a, &b);
    println!("Matrix multiplication completed");
}
